compiler/rustc_lint/src/errors.rs

```diff
@@ -0,0 +1,162 @@
+use rustc_errors::{fluent, AddSubdiagnostic, ErrorGuaranteed};
+use rustc_macros::{SessionDiagnostic, SessionSubdiagnostic};
+use rustc_session::{lint::Level, parse::ParseSess, SessionDiagnostic};
+use rustc_span::{Span, Symbol};
+
+#[derive(SessionDiagnostic)]
+#[diag(lint::overruled_attribute, code = "E0453")]
+pub struct OverruledAttribute {
+    #[primary_span]
+    pub span: Span,
+    #[label]
+    pub overruled: Span,
+    pub lint_level: String,
+    pub lint_source: Symbol,
+    #[subdiagnostic]
+    pub sub: OverruledAttributeSub,
+}
+//
+pub enum OverruledAttributeSub {
```

Comment:
Why can't `#[derive(SessionSubdiagnostic)]` be used here? I assume the `Option<Symbol>`?

Reply:
Yes, `#[derive(SessionSubdiagnostic)]` doesn't work on the `Option<Symbol>`.

Reply:
Okay, that's a shame. We should work out a better solution for stuff like `Option<Symbol>`, it would be rather helpful.